processing/src/main/java/org/apache/carbondata/processing/util/CarbonDataProcessorUtil.java

```diff
@@ -381,7 +381,9 @@ public static boolean isHeaderValid(String tableName, String[] csvHeader,
     List<CarbonDimension> dimensions =
         schema.getCarbonTable().getDimensionByTableName(factTableName);
     for (CarbonDimension dimension : dimensions) {
-      columnNames.add(dimension.getColName());
+      if (dimension.getSchemaOrdinal() != -1) {
```

Comment:
why this not -1 check is required because if any dimension added in schema it will have ordinal